internetarchive/cli/ia_upload.py

```diff
@@ -160,19 +163,22 @@ def main(argv, session):
             sys.exit(1)
 
     # Status check.
+    if args['<identifier>']:
+        item = session.get_item(args['<identifier>'])
```

Comment:
Do we really want to `get` an item that could be 1TB or more _before_ we do a status-check?

Reply:
Well, we need the `Item` object for both the size status check and the actual upload. While this structure means we needlessly fetch the item metadata when S3 is overloaded, it avoids more complicated conditions (e.g. first run the S3 overload check if `--status-check` is present, then fetch item metadata if the identifier is present, then check the item size if both are present, then exit successfully if `--status-check` is present), which in my opinion leads to less readable code. Alternatives are two lines with `get_item` calls, which is just as ugly, or some sort of lazy evaluation, which is somewhat complicated to implement. So I found this to be the least awkward solution.